buildscripts/autogen_sources.py

```diff
@@ -0,0 +1,155 @@
+    'ge': '>=',
+    'ne': '!=',
+    'eq': '==',
+}
+
+target_rel_filename = 'sdc/datatypes/hpat_pandas_series_autogenerated.py'
+
+
+if __name__ == '__main__':
+    sdc_src = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    abs_file_name = os.path.join(sdc_src, target_rel_filename)
+
+    # read templates_module as text and extract import section to be copied into auto-generated target file
+    module_text = inspect.getsource(templates_module)
+    module_text_lines = module_text.splitlines(keepends=True)
+    imports_line_numbers = [k for k, line in enumerate(module_text_lines) if 'import ' in line]
+    imports_start_line, import_end_line = min(imports_line_numbers), max(imports_line_numbers)
+    import_section_text = ''.join(module_text_lines[imports_start_line: import_end_line + 1])
+
+    # read function templates for arithmetic and comparison operators from tempaltes module
+    func_template_arithmetic_binop = inspect.getsource(templates_module.sdc_pandas_series_operator_binop)
+    func_template_comparison_binop = inspect.getsource(templates_module.sdc_pandas_series_operator_comp_binop)
+
+    exit_status = -1
+    try:
+        # open the target file for writing and do the main work
+        with open(abs_file_name, 'w', newline='') as file:
+            file.write(encoding_info)
+            file.write('\n')
+            file.write(copyright_header)
+            file.write('\n')
+            file.write(docstring_header)
+            file.write('\n')
+            file.write(import_section_text)
+
+            # certaing modifications are needed to be applied for templates, so
+            # verify correctness of produced code manually
+            for name in arithmetic_binops_symbols:
+                file.write('\n\n')
+                func_text = func_template_arithmetic_binop.replace('binop', name)
+                func_text = func_text.replace(' + ', ' ' + arithmetic_binops_symbols[name] + ' ')
+                func_text = func_text.replace('def ', '@overload(operator.{})\ndef '.format(name), 1)
+                file.write(func_text)
+
+            for name in comparison_binops_symbols:
+                file.write('\n\n')
+                func_text = func_template_comparison_binop.replace('comp_binop', name)
+                func_text = func_text.replace(' < ', ' ' + comparison_binops_symbols[name] + ' ')
+                func_text = func_text.replace('def ', '@overload(operator.{})\ndef '.format(name), 1)
+                file.write(func_text)
+
+            exit_status = 0
+
+    except Exception as e:
+        print('Exception of type {}: {} \nwhile writing to a file: {}\n'.format(
+                        type(e).__name__, e, os.path.normpath(abs_file_name)),
+                    file=sys.stderr)
+        exit_status = 1
+
+    if not exit_status:
+        print('Auto-generation sctipt completed successfully')
+    else:
+        print('Auto-generation failed, exit_status: {}'.format(exit_status))
```

Comment:
You could print these messages where you assign `exit_status` without `if-else` block.

Reply:
@densmirn Well, yes, but I think it's better to have this if-else as a separate finalization block at the end rather than scatter these prints across script body. I maybe wrong, but let's leave this as-is for now.

Reply:
Ok, let's leave this as is.